examples/api/test/widgets/actions/action_listener.0_test.dart

```diff
@@ -0,0 +1,49 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_api_samples/widgets/actions/action_listener.0.dart' as example;
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets('It should enable the action listener, trigger it and disable it', (WidgetTester tester) async {
+    final List<String?> log = <String?>[];
+
+    final DebugPrintCallback originalDebugPrint = debugPrint;
+    debugPrint = (String? message, {int? wrapWidth}) {
+      log.add(message);
+    };
+
+    await tester.pumpWidget(
+      const example.ActionListenerExampleApp(),
+    );
+
+    expect(find.widgetWithText(AppBar, 'ActionListener Sample'), findsOne);
+    expect(find.widgetWithText(OutlinedButton, 'Enable'), findsOne);
+
+    await tester.tap(find.byType(OutlinedButton));
+    await tester.pump();
+
+    expect(find.widgetWithText(OutlinedButton, 'Disable'), findsOne);
+    expect(find.widgetWithText(ElevatedButton, 'Call Action Listener'), findsOne);
+    expect(log, const <String?>['Action Listener was added']);
+
+    await tester.tap(find.text('Call Action Listener'));
+    await tester.pumpAndSettle();
+
+    expect(find.widgetWithText(SnackBar, 'Action Listener Called'), findsOne);
+
+    await tester.tap(find.text('Disable'));
+    await tester.pump();
+
+    expect(find.widgetWithText(OutlinedButton, 'Enable'), findsOne);
+    expect(
+      log,
+      const <String?>['Action Listener was added', 'Action Listener was removed'],
+    );
+
+      debugPrint = originalDebugPrint;
```

Comment:
To properly restore debugPrint, I think the current solution is to rely on try/finally.
See https://github.com/flutter/flutter/blob/b2b0a751ff70816f793f372b8d2776bc1a3263f3/packages/flutter/test/widgets/focus_manager_test.dart#L2257

Reply:
That's indeed safer ! Thank you ! 

I made the change in [refactor: Use try-finally](https://github.com/flutter/flutter/pull/150606/commits/576696e2e459556f8388c6bc4402ed9aa4996961)